### PR TITLE
Implement CargoFactory and filtering of events for delivery progress

### PR DESCRIPTION
**Pull request by orende**
This PR solves two TODOs related to Cargo: one in which the logic for creating a new Cargo instance is moved into a cargo factory class, and one in which events used in the `deriveDeliveryProgress` method are filtered so that only events for the specified cargo are used. Additionally, it removes the `DomainObjectUtils.nullSafe` method and replaces it with `ObjectUtils.defaultIfNull` from Apache Commons Lang 3.

### CODE REVIEW

**Review by mackapappa — COMMENTED** [review]
Looks good! Made a few comments...

File: src/main/java/se/citerus/dddsample/domain/model/cargo/Cargo.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ public Delivery delivery() {
    * @return The itinerary. Never null.
    */
   public Itinerary itinerary() {
-    return DomainObjectUtils.nullSafe(this.itinerary, Itinerary.EMPTY_ITINERARY);
+    return ObjectUtils.defaultIfNull(this.itinerary, Itinerary.EMPTY_ITINERARY);
```
[the remark this answers is not in the document]
Comment (orende) [reply src/main/java/se/citerus/dddsample/domain/model/cargo/Cargo.java:96]: I didn't actually know about `Objects.requireNonNullElse` but that's an even better replacement.

File: src/main/java/se/citerus/dddsample/domain/model/cargo/CargoFactory.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package se.citerus.dddsample.domain.model.cargo;
+
+import se.citerus.dddsample.domain.model.location.Location;
+import se.citerus.dddsample.domain.model.location.LocationRepository;
+import se.citerus.dddsample.domain.model.location.UnLocode;
+
+import java.util.Date;
+
+public class CargoFactory {
```
[the remark this answers is not in the document]
Comment (orende) [reply src/main/java/se/citerus/dddsample/domain/model/cargo/CargoFactory.java:9]: I didn't at the time, but looking at the code again I think it makes sense to turn it into a spring bean. There's no point in instantiating a new CargoFactory for each call to the method after all.  Will update the PR.

File: src/main/java/se/citerus/dddsample/domain/model/cargo/Cargo.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (orende) [reply src/main/java/se/citerus/dddsample/domain/model/cargo/Cargo.java:96]: I've added a ticket to refactor this to the Trello backlog.

**Pull request merged**